addons/mrp_subcontracting/tests/test_subcontracting.py

```diff
@@ -686,7 +603,85 @@ def test_flow_tracked_1(self):
         # Available quantities should be negative at the subcontracting location for each components
         avail_qty_comp1 = self.env['stock.quant']._get_available_quantity(self.comp1_sn, self.subcontractor_partner1.property_stock_subcontractor, allow_negative=True)
         avail_qty_comp2 = self.env['stock.quant']._get_available_quantity(self.comp2, self.subcontractor_partner1.property_stock_subcontractor, allow_negative=True)
-        avail_qty_finished = self.env['stock.quant']._get_available_quantity(self.finished_lot, wh.lot_stock_id)
+        avail_qty_finished = self.env['stock.quant']._get_available_quantity(self.finished_product, wh.lot_stock_id)
         self.assertEqual(avail_qty_comp1, -1)
         self.assertEqual(avail_qty_comp2, -1)
         self.assertEqual(avail_qty_finished, 1)
+
+    def test_flow_tracked_backorder(self):
+        """ This test uses tracked (serial and lot) component and tracked (serial) finished product """
+        todo_nb = 4
+        self.comp2.tracking = 'lot'
+        self.finished_product.tracking = 'serial'
+
+        # Create a receipt picking from the subcontractor
+        picking_form = Form(self.env['stock.picking'])
+        picking_form.picking_type_id = self.env.ref('stock.picking_type_in')
+        picking_form.partner_id = self.subcontractor_partner1
+        with picking_form.move_ids_without_package.new() as move:
+            move.product_id = self.finished_product
+            move.product_uom_qty = todo_nb
+        picking_receipt = picking_form.save()
+        picking_receipt.action_confirm()
+
+        # We should be able to call the 'record_components' button
+        self.assertTrue(picking_receipt.display_action_record_components)
+
+        # Check the created manufacturing order
+        mo = self.env['mrp.production'].search([('bom_id', '=', self.bom_tracked.id)])
+        self.assertEqual(len(mo), 1)
+        self.assertEqual(len(mo.picking_ids), 0)
+        wh = picking_receipt.picking_type_id.warehouse_id
+        self.assertEqual(mo.picking_type_id, wh.subcontracting_type_id)
+        self.assertFalse(mo.picking_type_id.active)
+
+        lot_comp2 = self.env['stock.production.lot'].create({
+            'name': 'lot_comp2',
+            'product_id': self.comp2.id,
+            'company_id': self.env.company.id,
+        })
+        serials_finished = []
+        serials_comp1 = []
+        for i in range(todo_nb):
+            serials_finished.append(self.env['stock.production.lot'].create({
+                'name': 'serial_fin_%s' % i,
+                'product_id': self.finished_product.id,
+                'company_id': self.env.company.id,
+            }))
+            serials_comp1.append(self.env['stock.production.lot'].create({
+                'name': 'serials_comp1_%s' % i,
+                'product_id': self.comp1_sn.id,
+                'company_id': self.env.company.id,
+            }))
+
+        # TODO : Waiting fix for (2, 0) in test form fix
```

Comment:
was the fix merged ?

Reply:
In the testing environment, no :/

Reply:
My bad, it is merged in https://github.com/odoo/odoo/pull/54642/commits/fae2ecd580ff2a9c67de25ea04378858a4633a8f
I will uncomment this part